tests/components/webostv/test_notify.py

```diff
@@ -0,0 +1,121 @@
+    assert hass.services.has_service(NOTIFY_DOMAIN, TV_NAME)
+
+    await hass.services.async_call(
+        NOTIFY_DOMAIN,
+        TV_NAME,
+        {
+            ATTR_MESSAGE: MESSAGE,
+            CONF_SERVICE_DATA: {
+                CONF_ICON: ICON_PATH,
+            },
+        },
+        blocking=True,
+    )
+    assert client.mock_calls[0] == call.connect()
+    assert client.connect.call_count == 1
+    client.send_message.assert_called_with(MESSAGE, icon_path=ICON_PATH)
+
+
+async def test_notify_not_connected(hass, client, monkeypatch):
+    """Test sending a message when client is not connected."""
+    await setup_webostv(hass, "fake-uuid")
+    assert hass.services.has_service(NOTIFY_DOMAIN, TV_NAME)
+
+    monkeypatch.setattr(client, "is_connected", Mock(return_value=False))
+    await hass.services.async_call(
+        NOTIFY_DOMAIN,
+        TV_NAME,
+        {
+            ATTR_MESSAGE: MESSAGE,
+            CONF_SERVICE_DATA: {
+                CONF_ICON: ICON_PATH,
+            },
+        },
+        blocking=True,
+    )
+    assert client.mock_calls[0] == call.connect()
+    assert client.connect.call_count == 2
+    client.send_message.assert_called_with(MESSAGE, icon_path=ICON_PATH)
+
+
+async def test_icon_not_found(hass, caplog, client, monkeypatch):
+    """Test notify icon not found error."""
+    await setup_webostv(hass, "fake-uuid")
+    assert hass.services.has_service(NOTIFY_DOMAIN, TV_NAME)
+
+    monkeypatch.setattr(client, "send_message", Mock(side_effect=FileNotFoundError))
+    await hass.services.async_call(
+        NOTIFY_DOMAIN,
+        TV_NAME,
+        {
+            ATTR_MESSAGE: MESSAGE,
+            CONF_SERVICE_DATA: {
+                CONF_ICON: ICON_PATH,
+            },
+        },
+        blocking=True,
+    )
+    assert client.mock_calls[0] == call.connect()
+    assert client.connect.call_count == 1
+    client.send_message.assert_called_with(MESSAGE, icon_path=ICON_PATH)
+    assert f"Icon {ICON_PATH} not found" in caplog.text
+
+
+@pytest.mark.parametrize(
+    "side_effect,error",
+    [
+        (WebOsTvPairError, "Pairing with TV failed"),
+        (ConnectionRefusedError, "TV unreachable"),
+    ],
+)
+async def test_connection_errors(hass, caplog, client, monkeypatch, side_effect, error):
+    """Test connection errors scenarios."""
+    await setup_webostv(hass, "fake-uuid")
+    assert hass.services.has_service("notify", TV_NAME)
+
+    monkeypatch.setattr(client, "is_connected", Mock(side_effect=side_effect))
+    await hass.services.async_call(
+        NOTIFY_DOMAIN,
+        TV_NAME,
+        {
+            ATTR_MESSAGE: MESSAGE,
+            CONF_SERVICE_DATA: {
+                CONF_ICON: ICON_PATH,
+            },
+        },
+        blocking=True,
+    )
+    assert client.mock_calls[0] == call.connect()
+    assert client.connect.call_count == 1
+    client.send_message.assert_not_called()
+    assert error in caplog.text
+
+
+async def test_no_discovery_info(hass):
```

Comment:
This tests make sure we go through this code without errors:
https://github.com/home-assistant/core/blob/deb32c129ef4c945c359c5b13b86f926e9bd46d9/homeassistant/components/webostv/notify.py#L26-L27
Since we should not do anything it just tests that the notify platform loaded without errors.
I am not sure if there is a possible case for discovery info to be `None` so maybe we can just assert in the original code